### PR TITLE
Restricted Gasoline Zeds to Gas Stations (w/ rare chance)

### PR DESCRIPTION
**Pull request by sonphantrung**
#### Summary
Balance "Restricted Gasoline Zeds to Gas Stations"

#### Purpose of change
Gasoline zombies are currently quite problematic:
- Appears everywhere, even under the most nonsensical places (labs for example)
- Occasionally "hurt" itself offscreen, explode into fireballs, ruining loots, giving players nothing but anger.
This solves those

#### Describe the solution
- Removed gasoline zeds from zombies' evolution
- Added it as a rare, late spawned (`"starts": "7 days"`) gas station zombie, suggested that it has consumed LOTS OF GAS there

#### Describe alternatives you've considered
Not doing this, and seeing more players complaining about the BS of gas zeds

#### Testing
If it doesn't throw errors, it should be good

#### Additional context
For me, playing w/ blacklist mods is considered a weak-minded, last ditch solution to undesired zeds

### CODE REVIEW

**Review by bombasticSlacks — APPROVED** [review]
I'm not the only one with a voice on this but I think this is a reasonable solution to the odd interactions gasoline zombies cause when paired with the reality bubble.

File: data/json/mapgen/gas_stations/s_gas.json [hunk 1]
```diff
@@ -49,7 +49,8 @@
       "place_liquids": [
         { "liquid": "gasoline", "x": 2, "y": 3, "repeat": [ 200, 1075 ] },
         { "liquid": "diesel", "x": 4, "y": 3, "repeat": [ 200, 1075 ] }
-      ]
+      ],
+      "place_monsters": [ { "monster": "GROUP_ZOMBIE_GAS", "x": [ 5, 10 ], "y": [ 6, 10 ], "chance": 2 } ]
```
Comment (LovamKicsiGazsii) [inline data/json/mapgen/gas_stations/s_gas.json:53]: `place_monsters` calculates chance a bit differently. Here, `chance` represents a 1 in N chance that the entire group will spawn (according to the mapgen doc). This means gasoline zeds have a 50% chance of spawning.

**Comment by Inglonias** [comment]
If merged, this would close #56862

File: data/json/mapgen/gas_stations/s_gas.json [hunk 2]
```diff
@@ -254,7 +256,8 @@
         { "item": "magazines", "x": [ 9, 9 ], "y": [ 13, 15 ], "chance": 90, "repeat": [ 1, 25 ] },
         { "item": "softdrugs", "x": [ 8, 8 ], "y": [ 13, 15 ], "chance": 80, "repeat": [ 1, 8 ] }
       ],
-      "place_item": [ { "item": "shotgun_s", "x": 3, "y": 20 }, { "item": "shot_00", "x": 7, "y": 19, "chance": 50 } ]
+      "place_item": [ { "item": "shotgun_s", "x": 3, "y": 20 }, { "item": "shot_00", "x": 7, "y": 19, "chance": 50 } ],
+      "place_monsters": [ { "monster": "GROUP_ZOMBIE_GAS", "x": [ 5, 10 ], "y": [ 6, 10 ], "chance": 2 } ]
```
Comment (Inglonias) [inline data/json/mapgen/gas_stations/s_gas.json:260]:
```suggestion
      "place_monster": [ { "group": "GROUP_ZOMBIE_GAS", "x": [ 5, 10 ], "y": [ 6, 10 ], "repeat": 5, "chance": 2 } ]
```

**Review by Inglonias — CHANGES_REQUESTED** [review]
Strongly advise avoiding `place_monsters` flag, as it will place every entry in the group once if that chance procs. Use `place_monster` with the `repeat` field instead.

**Comment by LyranRenegade** [comment]
My 2 cts, don't nerf stuff too extremely just because of a potential ruined run.
I mean, I dont have any problems with the gas-z's and have a lot of fun using them tactically against their brethren.
Now, I'm sure there are cases you'd wish they weren't there, but hey..

**Pull request merged**